mungegithub/mungers/assign-reassign-handler.go

```diff
@@ -62,40 +63,54 @@ func (AssignReassignHandler) Initialize(config *github.Config, features *feature
 func (AssignReassignHandler) EachLoop() error { return nil }
 
 // AddFlags will add any request flags to the cobra `cmd`
-func (AssignReassignHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+func (h AssignReassignHandler) AddFlags(cmd *cobra.Command, config *github.Config) {
+	cmd.Flags().BoolVar(&h.checkValid, "check-valid-reviewer", true, "Flag indicating whether to allow any one to be assigned or just OWNERs files reviewers")
+}
 
 // Munge is the workhorse the will actually make updates to the PR
 func (h AssignReassignHandler) Munge(obj *github.MungeObject) {
 	if !obj.IsPR() {
 		return
 	}
-	reviewers := getReviewers(obj)
-	if len(reviewers) == 0 {
-		return
-	}
 
 	comments, err := obj.ListComments()
 	if err != nil {
 		glog.Errorf("unexpected error getting comments: %v", err)
 		return
 	}
 
-	h.assignOrRemove(obj, comments, reviewers)
+	toAssign, toUnassign, err := h.assignOrRemove(obj, comments, true)
+	if err != nil {
+		return
+	}
+	//assign and unassign reviewers as necessary
+	for _, username := range toAssign.List(){
+		obj.AssignPR(username)
+	}
+	if len(toUnassign) > 0 {
+		is := goGithub.IssuesService{}
+		is.RemoveAssignees(OWNER, *obj.Issue.Repository.Name, *obj.Issue.Number, toUnassign.List())
+	}
 }
 
 //assignOrRemove checks to see when someone comments "/assign" or "/reassign"
 // "/assign" self assigns the PR
 // "/reassign" unassignes the commenter and reassigns to someone else
 // [TODO] "/reassign <github handle>" reassign to this person
-func (h *AssignReassignHandler) assignOrRemove(obj *github.MungeObject, comments []*goGithub.IssueComment, reviewers mungerutil.UserSet) {
-
-	fileList, err := obj.ListFiles()
-	if err != nil {
-		glog.Error("Could not list the files for PR %v", obj.Issue.Number)
-		return
+func (h *AssignReassignHandler) assignOrRemove(obj *github.MungeObject, comments []*goGithub.IssueComment, checkValid bool) (toAssign, toUnassign sets.String, _ error){
+
+	toAssign = sets.String{}
+	toUnassign = sets.String{}
+	potential_owners := weightMap{}
+	if checkValid {
```

Comment:
I assume this is for test, kind of gross.
Maybe more easy to test: replace checkValid parameter with list of files parameter. Then the test or the actual code can use a different way to generate the list of files.